Sources/Combine/Extensions/Publisher+Utility.swift

```diff
@@ -12,32 +12,148 @@ import Combine
 @available(iOS 13, *)
 public extension Publisher {
 
-    /// Maps each sequence element to given value.
+    /// Erases the Publisher's error type
     ///
-    /// - Parameter value: Value to map
-    /// - Returns: Sequence where all elements are given value.
-    func mapTo<T>(_ value: T) -> Publishers.Map<Self, T> {
-        return map { _ in value }
+    /// - Returns: Publisher with a Failure type of Error
+    func eraseError() -> Publishers.MapError<Self, Error> {
+        return mapError { $0 as Error }
+    }
+
+    /// Catches the error converting it into the specified value
+    ///
+    /// - Returns: Publisher with the specified value
+    func catchErrorReturn<T>(_ value: T) -> Publishers.Catch<Self, Just<T>> {
+        return self.catch { _ in Just(value) }
+    }
+
+    /// Debounces the Publisher for a specified amount of time on the main queue
+    ///
+    /// - Returns: Publisher with debounce for a specified amount of time
+    func debounce(for dueTime: DispatchQueue.SchedulerTimeType.Stride) -> Publishers.Debounce<Self, DispatchQueue> {
+        return debounce(for: dueTime, scheduler: .main)
     }
 
     /// Maps each sequence element to Void type.
     ///
     /// - Returns: Sequence where all elements are of Void type.
     func mapToVoid() -> Publishers.Map<Self, Void> {
-        return mapTo(())
-    }
-
-    func inclusivePrefix(while predicate: @escaping (Self.Output) -> Bool) -> Publishers.HandleEvents<Publishers.PrefixWhile<Self>> {
-        var previousValue: Output?
-        return self
-            .prefix(while: { output in
-                guard let previousValue = previousValue else {
-                    return predicate(output)
-                }
-                return predicate(previousValue)
-            })
-            .handleEvents(receiveOutput: {
-                previousValue = $0
-            })
+        return mapToValue(())
+    }
+
+    /// Prefixes elements of a sequence for the given `predicate` while also including the value at which the `prefix` operator stops.
+    func inclusivePrefix(while predicate: @escaping (Self.Output) -> Bool) -> AnyPublisher<Self.Output, Failure> {
+        let shared = share(replay: 1)
+        // In order to get a prefix that's inclusive, we need to make use of the `share(replay:)` operator
+        // First, we prefix as usual until we get all of the elements we want, for the given predicate.
+        // Next, we make use of the replayed sequence to get the first value which doesn't fulfill the predicate requirement.
+        return shared
+            .prefix(while: predicate)
+            .append(shared.first { !predicate($0) })
+            .eraseToAnyPublisher()
```

Comment:
I would use [scan](https://developer.apple.com/documentation/combine/fail/scan(_:_:)) to handle this. No need to share in that case 😄

Reply:
The code would be `self.scan(...).prefix(...).map(...)`